trieste/models/keras/models.py

```diff
@@ -367,4 +371,321 @@ def __setstate__(self, state: dict[str, Any]) -> None:
+    in the early training of the error predictor. The warm start is enabled by default, and can be disabled by
+    setting the ``init_buffer`` argument to False.
+
+    Note that currently we do not support setting up the model with dictionary configs and saving
+    the model during Bayesian optimization loop (``track_state`` argument in
+    :meth:`~trieste.bayesian_optimizer.BayesianOptimizer.optimize` method should be set to `False`).
+    """
+    def __init__(
+        self,
+        model: Sequence[dict[str, Any]],
+        optimizer: Optional[KerasOptimizer] = None,
+        init_buffer: bool = False
+    ) -> None:
+
+        """
+        :param model: A dictionary with two models: the main predictor and the auxiliary error model.
+            The main Keras model should be a compiled model of class :class:`trieste.models.DeepEnsemble`
+            or class :class:`trieste.models.keras.MonteCarloDropout`. The auxiliary model is an instance of
+            :class:`trieste.models.keras.EpistemicUncertaintyNetwork`. The model has to be built but not
+            compiled.
+        :param optimizer: The optimizer wrapper with necessary specifications for compiling and
+            training the model. Defaults to :class:`~trieste.models.optimizer.KerasOptimizer` with
+            :class:`~tf.optimizers.Adam` optimizer, mean squared error loss and a dictionary
+            of default arguments for Keras `fit` method: 1000 epochs, batch size 16, early stopping
+            callback with patience of 50, and verbose 0. 
+            See https://keras.io/api/models/model_training_apis/#fit-method for a list 
+            of possible arguments.
+        :param init_buffer: A boolean that enables the pre-training of the error predictor by training several
+            main models using cross-validated slices of the dataset and fitting the resulting models on
+            the entire dataset. This constructs a dataset of [N*K, D] observations nad squared losses, which
+            are used to train the auxiliary predictor.
+        """
+
+        super().__init__(optimizer)
+
+        if not self.optimizer.fit_args:
+            self.optimizer.fit_args = {
+                "verbose": 0,
+                "epochs": 1000,
+                "batch_size": 32,
+                "callbacks": [
+                    tf.keras.callbacks.EarlyStopping(
+                        monitor="loss", patience=100, restore_best_weights=True
+                    )
+                ],
+            }
+
+        if self.optimizer.loss is None:
+            self.optimizer.loss = negative_log_likelihood
+
+        self._learning_rate = self.optimizer.optimizer.learning_rate.numpy()
+        
+        model["e_model"].compile(
+            self.optimizer.optimizer,
+            loss="mse",
+            metrics=[self.optimizer.metrics],
+        )
+
+        self._model = model
+        self._init_buffer = init_buffer
+        self._buffer_iter = 1
```

Comment:
In fact, the idea would be to have a single integer parameter: 0 (default) is no initial buffer, otherwise any number is the number of full loops. I didn't change it out of just how much I have used the init_buffer param everywhere (I added it y'day to include it as a models_test), but you make a good point.

Reply:
I changed it.